### PR TITLE
Optical flow fixes for ModalAI thermal optic flow

### PR DESCRIPTION
**Pull request by katzfey**
### Solved Problem
Some increased timeouts and other changes to support ModalAI thermal optic flow with Flir Lepton

### CODE REVIEW

File: src/modules/ekf2/EKF/optical_flow_control.cpp [hunk 1]
```diff
@@ -204,7 +204,7 @@ void Ekf::controlOpticalFlowFusion(const imuSample &imu_delayed)
 			if (_time_delayed_us > (_flow_sample_delayed.time_us - uint32_t(1e6f * _flow_sample_delayed.dt) / 2)) {
 				// Fuse optical flow LOS rate observations into the main filter only if height above ground has been updated recently
 				// but use a relaxed time criteria to enable it to coast through bad range finder data
-				if (isRecent(_time_last_hagl_fuse, (uint64_t)10e6)) {
+				if (isRecent(_time_last_hagl_fuse, (uint64_t)100e6)) {
```
Comment (dagar) [inline src/modules/ekf2/EKF/optical_flow_control.cpp:207]: drop for now?

File: src/modules/ekf2/EKF/optflow_fusion.cpp [hunk 2]
```diff
@@ -77,6 +77,19 @@ void Ekf::updateOptFlow(estimator_aid_source2d_s &aid_src)
...
+
+	const Vector24f state_vector = getStateAtFusionHorizonAsVector();
+
+	Vector2f innov_var;
+	Vector24f H;
+	sym::ComputeFlowXyInnovVarAndHx(state_vector, P, range, R_LOS, FLT_EPSILON, &innov_var, &H);
+	innov_var.copyTo(_aid_src_optical_flow.innovation_variance);
+
+	// run the innovation consistency check and record result
+	setEstimatorAidStatusTestRatio(_aid_src_optical_flow, math::max(_params.flow_innov_gate, 1.f));
```
Comment (dagar) [inline src/modules/ekf2/EKF/optflow_fusion.cpp:89]:
```suggestion
	setEstimatorAidStatusTestRatio(aid_src, math::max(_params.flow_innov_gate, 1.f));
```

File: src/modules/ekf2/EKF/optflow_fusion.cpp [hunk 3]
```diff
@@ -77,6 +77,19 @@ void Ekf::updateOptFlow(estimator_aid_source2d_s &aid_src)
 	const float R_LOS = calcOptFlowMeasVar(_flow_sample_delayed);
 	aid_src.observation_variance[0] = R_LOS;
 	aid_src.observation_variance[1] = R_LOS;
+
+	const Vector24f state_vector = getStateAtFusionHorizonAsVector();
+
+	Vector2f innov_var;
+	Vector24f H;
+	sym::ComputeFlowXyInnovVarAndHx(state_vector, P, range, R_LOS, FLT_EPSILON, &innov_var, &H);
+	innov_var.copyTo(_aid_src_optical_flow.innovation_variance);
```
[the remark this answers is not in the document]
Comment (dagar) [reply src/modules/ekf2/EKF/optflow_fusion.cpp:86]: updateOptFlow() is called for both regular optical flow fusion and the terrain estimator flow fusion

File: src/modules/ekf2/EKF/optflow_fusion.cpp [hunk 4]
```diff
@@ -77,6 +77,19 @@ void Ekf::updateOptFlow(estimator_aid_source2d_s &aid_src)
...
+
+	// run the innovation consistency check and record result
+	setEstimatorAidStatusTestRatio(_aid_src_optical_flow, math::max(_params.flow_innov_gate, 1.f));
+
+	_innov_check_fail_status.flags.reject_optflow_X = (_aid_src_optical_flow.test_ratio[0] > 1.f);
```
Comment (dagar) [inline src/modules/ekf2/EKF/optflow_fusion.cpp:91]:
```suggestion
```

File: src/modules/ekf2/EKF/optflow_fusion.cpp [hunk 5]
```diff
@@ -77,6 +77,19 @@ void Ekf::updateOptFlow(estimator_aid_source2d_s &aid_src)
...
+	_innov_check_fail_status.flags.reject_optflow_Y = (_aid_src_optical_flow.test_ratio[1] > 1.f);
```
Comment (dagar) [inline src/modules/ekf2/EKF/optflow_fusion.cpp:92]:
```suggestion
```

File: src/modules/ekf2/EKF/optical_flow_control.cpp [hunk 6]
```diff
@@ -59,7 +59,7 @@ void Ekf::controlOpticalFlowFusion(const imuSample &imu_delayed)
 
 	// Accumulate autopilot gyro data across the same time interval as the flow sensor
 	const Vector3f delta_angle(imu_delayed.delta_ang - (getGyroBias() * imu_delayed.delta_ang_dt));
-	if (_delta_time_of < 0.1f) {
+	if (_delta_time_of < 0.5f) {
```
Comment (dagar) [inline src/modules/ekf2/EKF/optical_flow_control.cpp:62]: This just seems really high, maybe we start with doubling it (consistent with the max below)?

File: src/modules/ekf2/EKF/optical_flow_control.cpp [hunk 7]
```diff
@@ -75,7 +75,7 @@ void Ekf::controlOpticalFlowFusion(const imuSample &imu_delayed)
 		const bool is_tilt_good = (_R_to_earth(2, 2) > _params.range_cos_max_tilt);
 
 		const float delta_time_min = fmaxf(0.7f * _delta_time_of, 0.001f);
-		const float delta_time_max = fminf(1.3f * _delta_time_of, 0.2f);
+		const float delta_time_max = fminf(1.3f * _delta_time_of, 0.5f);
```
Comment (dagar) [inline src/modules/ekf2/EKF/optical_flow_control.cpp:78]:
```suggestion
		const float delta_time_max = fminf(1.3f * _delta_time_of, 0.2f);
```

File: src/modules/ekf2/EKF/optflow_fusion.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (AlexKlimaj) [reply src/modules/ekf2/EKF/optflow_fusion.cpp:91]: It looks like updateOptFlow(_aid_src_optical_flow); gets called and this is updated in controlOpticalFlowFusion

**Comment by katzfey** [comment]
This one is no longer needed. It was pulled in as part of 21769

**Pull request closed without merging**